### PR TITLE
Fix 'fmt' for Mac OS users

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitalii Parfonov <vparfono@redhat.com>

### Description

`-r` (`--no-run-if-empty`)  disables execution if output is empty,
remove '-r' from xargs options, output will never be empty

/cc 
/assign 

/cherry-pick 

### Links

- Depending on PR(s): N/A
- Bugzilla: N/A
- Github issue: N/A
- JIRA: N/A
- Enhancement proposal: N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Define the target to run if make is called with no arguments.
 default: check
+UNAME_S := $(shell uname -s)
```
Comment (periklis) [inline Makefile:3]: This variable implies the `OS`, so let's call it so. Also a good practice is to lowercase everything `shell uname -s | tr '[A-Z]' '[a-z]'`

File: Makefile [hunk 2]
```diff
@@ -96,7 +97,13 @@ lint: $(GOLANGCI_LINT)
 
 fmt:
 	@echo gofmt		# Show progress, real gofmt line is too long
-	find pkg cmd test -name '*.go' | xargs -r gofmt -s -l -w
+	@if [ $(UNAME_S) == 'Linux' ] ; then \
+		find pkg cmd test -name '*.go' | xargs -r gofmt -s -l -w; \
+	fi
+	#BSD (MacOS) xargs does not have option -r (--no-run-if-empty)
+	@if [ $(UNAME_S) = 'Darwin' ] ; then \
+		find pkg cmd test -name '*.go' | xargs gofmt -s -l -w; \ #BSD xargs does not have this behavior, so there's no need to disable it.
```
Comment (periklis) [inline Makefile:105]: If the whole fix is about `xargs`, could you check if you could use the one from brew?

**Review by periklis — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (vparfonov) [reply Makefile]: I think is nothing to fix in `xargs` side, just different behavior

File: Makefile [hunk 3]
```diff
@@ -96,7 +97,13 @@ lint: $(GOLANGCI_LINT)
...
 fmt:
 	@echo gofmt		# Show progress, real gofmt line is too long
-	find pkg cmd test -name '*.go' | xargs -r gofmt -s -l -w
+	@if [ $(OS_NAME) == 'Linux' ] ; then \
+		find pkg cmd test -name '*.go' | xargs -r gofmt -s -l -w; \
+	fi
+	# BSD (MacOS) xargs does not have option -r (--no-run-if-empty)
+	@if [ $(OS_NAME) = 'Darwin' ] ; then \
+		find pkg cmd test -name '*.go' | xargs gofmt -s -l -w; \
+	fi
```
Comment (periklis) [inline Makefile:106]: The checks should use now lowercase, too, e.g. `linux`.

**Review by periklis — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (periklis) [reply Makefile]: If you use the xargs from homebrew aka from GNU, you should get the same behavior right?

File: Makefile (hunk 3 above)
Comment (vparfonov) [reply Makefile]: done

File: Makefile [hunk 4]
```diff
@@ -96,7 +97,13 @@ lint: $(GOLANGCI_LINT)
 
 fmt:
 	@echo gofmt		# Show progress, real gofmt line is too long
-	find pkg cmd test -name '*.go' | xargs -r gofmt -s -l -w
+	@if [ $(OS_NAME) == 'linux' ] ; then \
+		find pkg cmd test -name '*.go' | xargs -r gofmt -s -l -w; \
+	fi
+	# BSD (MacOS) xargs does not have option -r (--no-run-if-empty)
+	@if [ $(OS_NAME) = 'darwin' ] ; then \
+		find pkg cmd test -name '*.go' | xargs gofmt -s -l -w; \
```
Comment (alanconway) [inline Makefile:105]: Actually we can just delete the -r, it'll never be empty (or if it is, somebody's deleted the entire code-base, which probably deserves to be an error!) I was coming from writing a different script where the -r was needed and I put it by reflex.

**Review by alanconway — CHANGES_REQUESTED** [review]

File: Makefile (hunk 4 above)
Comment (vparfonov) [reply Makefile:105]: thx

**Review by periklis — COMMENTED** [review]
LGTM, please squash the commits.

**Pull request merged**